pycqed/analysis_v2/base_analysis.py

```diff
@@ -138,11 +140,14 @@ def __init__(self, t_start: str = None, t_stop: str = None,
         # These options determine what data to extract #
         ################################################
         scan_label = self.options_dict.get('scan_label', label)
+        if scan_label is None:
```

Comment:
Would be much better to instead replace the default argument with `''`, a hidden if statement that replaces some parameter is quite horrible for debugging.

Reply:
The default argument is indeed a str='', see definition of the __init__ of the base_class.

Reply:
if you agree, could you remove this addition?

Reply:
Tests fail without it. 
Cleanup of this should be addressed in a different pull request. So I'm going to change it back.

Reply:
Culprit is the default label in the spec analysis. This is now changed, and the if statement removed.
So should be all good now.